services/user.js

```diff
@@ -105,6 +107,15 @@ const userService = {
     }
   },
 
+  updateUserProfile: async (id, updateData) => {
+    const filteredUpdateData = filterAllowedFields(updateData, allowedUserFieldsForUpdate)
+    const user = await User.findByIdAndUpdate(id, filteredUpdateData, { new: true }).exec()
```

Comment:
don't we need lean() here?

Reply:
and why we need { new: true } if this service doesn't return any data?

Reply:
we need return cause there is checking for user existing

Reply:
oh, i have got what are u talking about, new: true is really useless in this service